include/cell/copy/copy_atom.hpp

```diff
@@ -60,58 +60,232 @@ struct LoadMatBase {
     }
 };
 
-template <typename DType_, const int kSegRows_, const int kSegCols_,
-          const int kRstride_, const int kCstride_, const size_t kBitPerAccess>
+template <typename Shared, const tl::Layout kType, const size_t kBitPerAccess>
 struct BaseTileStorer;
 
-template <typename DType_, const int kSegRows_, const int kSegCols_,
-          const int kRstride_, const int kCstride_>
-struct BaseTileStorer<DType_, kSegRows_, kSegCols_, kRstride_, kCstride_, 32> {
-    using DType = DType_;
-    static constexpr int kSegRows = kSegRows_;
-    static constexpr int kSegCols = kSegCols_;
-    static constexpr int kRstride = kRstride_;
-    static constexpr int kCstride = kCstride_;
+/// TODO(haruhi): try to reduece reusable codes.
+template <typename Shared>
+struct BaseTileStorer<Shared, tl::Layout::kRowMajor, 16> {
+    using DType = Shared::DType;
 
     DEVICE void operator()(const DType* src_, DType* dst_) {
         const int* src = reinterpret_cast<const int*>(src_);
         int* dst = reinterpret_cast<int*>(dst_);
 
+        int lane_row = lane_row_id();
+        int lane_col = lane_col_id();
+
+        // A base tile has a fixed shape of 16x16 (a 16x16 2D coordinate space
+        // with integer indices ranging from 0 to 255). `row` and `col` are used
+        // to calculate the index of an element within this 16x16 coordinate
+        // space.
+        int row = 0, col = 0;
 #pragma unroll
-        for (int i = 0; i < kSegCols; ++i) {
+        for (int i = 0; i < kSegRows; ++i) {
+            row = lane_row + i * tl::num_rows<ThreadLayout>;
 #pragma unroll
-            for (int j = 0; j < kSegRows; ++j) {
-                dst[i * kRstride + j * kCstride] = src[j * kSegCols + i];
+            for (int j = 0; j < kSegCols; ++j) {
+                col = (lane_col + j * tl::num_cols<ThreadLayout>)*kElemPerSeg;
```

Comment:
typo: `col = (lane_col + j * tl::num_cols<ThreadLayout>) * kElemPerSeg;`

Reply:
This line is automatically formatted by clang-format, but I'm not sure how to configure clang-format correctly. I chose another way to write it to bypass the problem.